### ISSUE TITLE
query-ask: length greater than remaining number of bytes in buffer

### ISSUE THREAD

**Issue by jsign** [filecoin-project/lotus#810]
```
~ % lotus client query-ask t01010
2019-12-09T11:17:31.451-0300	WARN	main	lotus/main.go:72	length greater than remaining number of bytes in buffer
~ % lotus client query-ask t01281 
2019-12-09T11:17:37.081-0300	WARN	main	lotus/main.go:72	length greater than remaining number of bytes in buffer
~ % lotus client query-ask t01271
```
For other miners `query-ask` works correctly.
Seems that the error bubbled up from [go-multihash](https://github.com/multiformats/go-multihash/blob/4b0a826bbb736a6b4c2e0b60c9a0fccb94ebf7c5/multihash.go#L314).

Running `lotus` from commit `dec458718ff4751`.

**Comment by jsign** [comment]
Adding here some more extra info mentioned by _resende_ [in slack](https://filecoinproject.slack.com/archives/CPFTWMY7N/p1576061400347200):
>and on the rpc i pass the peer ID ( that i get from the ID endpoint with the miner address and the tip set ) and the miner address and i get a
>`"{\"jsonrpc\":\"2.0\",\"id\":0,\"error\":{\"code\":-32700,\"message\":\"unmarshaling params for '%!s(func(*api.Fu
llNodeStruct, context.Context, peer.ID, address.Address) (*types.SignedStorageAsk, error)=0xb3fce0)': input isn't va
lid multihash\"}}\n"}}}  `

**Comment by jsign** [comment]
Jumped to see the root of the problem.

The original described problem is an error in the CLI side [here](https://github.com/filecoin-project/lotus/blob/master/cli/client.go#L303). The `StateCall` returns an invalid/empty `MessageReceipt.Result`  making `IDFromBytes` fail with the log message.

Digged into the chain to see the unmarshaled cbor for this `MinerInfo`:
```
actors.MinerInfo{Owner:[3 - 8b2fbe0e3af35a63c9b9866e10e57b2092ac8aa5155c7800fd1920b2499c4cc80a650d113c985cbe3664f49810d370d9], Worker:[3 - 8b2fbe0e3af35a63c9b9866e10e57b2092ac8aa5155c7800fd1920b2499c4cc80a650d113c985cbe3664f49810d370d9], PeerID:"SETME", SectorSize:0x800000000} 
```
So.. `SETME` makes pretty clear that `IDFromBytes` will fail.
Talking with @whyrusleeping , seems to indicate that the miner hasn't been initialized yet.

TL;DR:
- The output error may have a nicer message. Maybe considering the extra case.
- At least for the two problematic peers, both fail for this same reason. (e.g: `t01010` and `t01281`)

---

I bet what _resende_ mentioned in that message is a failed attempt to call directly the RPC... which is a different error by itself. Feels like a wrong `peerID`... so hard to know without his original request.
